atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SerializableTransactionManager.java

```diff
@@ -18,16 +18,26 @@
 
 import com.palantir.atlasdb.cleaner.Cleaner;
 import com.palantir.atlasdb.transaction.api.PreCommitCondition;
-import com.palantir.atlasdb.transaction.api.TransactionFailedRetriableException;
 import com.palantir.atlasdb.transaction.api.TransactionManager;
 import com.palantir.atlasdb.transaction.api.TransactionTask;
 
 public interface SerializableTransactionManager extends TransactionManager {
-    RawTransaction setupRunTaskWithConditionThrowOnConflict(PreCommitCondition condition);
+    /**
+     * Returns a {@link SerializableTransaction} that can be used for reads and writes.
+     * This behaves as the transactions returned in {@link TransactionManager#runTaskWithRetry(TransactionTask)}.
+     */
+    <C extends PreCommitCondition, E extends Exception> ServiceWriteTransaction getWriteTransaction(
+            C condition) throws E;
 
-    <T, E extends Exception> T finishRunTaskWithLockThrowOnConflict(RawTransaction tx,
-            TransactionTask<T, E> task)
-            throws E, TransactionFailedRetriableException;
+    /**
+     * Returns a transaction that can just be used for reads.
+     * Note that this transaction does not lock the immutable timestamp. Therefore, it is possible that sweep has
+     * deleted the values this transaction would read, causing correctness issues.
+     * The burden for locking the immutable timestamp, and preventing correctness problems is on the client.
```

Comment:
We should expand docs a bit more - the client should lock the immutable timestamp and also provide a lock checking precommit condition. If we want to help avoid clients misusing this, we could require a lock checking precommit condition as an argument.